src/shared/utils/download.ts

```diff
@@ -2,13 +2,19 @@ import { fileExtensionFromResourceEncoding } from '../../utils/contentTypes';
 
 export const download = (filename: string, mediaType: string, data: any) => {
   const blob = new Blob([data], { type: mediaType });
-  const extention = fileExtensionFromResourceEncoding(mediaType);
+  const extension = fileExtensionFromResourceEncoding(mediaType);
+
+  // Check if filename already ends with the correct extension
+  const hasCorrectExtension = filename.endsWith(`.${extension}`);
+
```

Comment:
I am afraid that some data has no `mediaType` and will have to rely on the `filename.ext` 
can you include that in case